examples/flutter_gallery/lib/demo/cupertino/cupertino_segmented_control_demo.dart

```diff
@@ -46,7 +46,13 @@ class _CupertinoSegmentedControlDemoState extends State<CupertinoSegmentedContro
     ),
   };
 
-  int sharedValue = 0;
+  final ValueNotifier<int> controller = ValueNotifier<int>(0);
+
+  @override
+  void initState() {
+    super.initState();
+    controller.addListener(() { setState(() {}); });
```

Comment:
I don't think we want to require developers to set up the segmented control this way. The convention for this kind of thing is to for the onValueChanged() callback set update the enclosing widget's state and call setState(). Conventionally, the controller parameter is optional. If the controller is specified then the caller must manage its lifetime.